### PR TITLE
[RichTextInput] add tests and fix bugs

### LINKED ISSUE #3206: [ra-input-rich-text] validation doesn't rerun on input change
**What you were expecting:**

I expect no validation error when required rich text input has content

**What happened instead:**

Validation error persists, even when input is not empty anymore

**Steps to reproduce:**

1. Run simple example
2. Go to PostCreate.js
3. Add `validate={[required()]}` to `RichTextInput`
4. Try to edit rich text input

**Other information:**

Screen-Recording-2019-05-09-at-15 47 35

Looking at source code, there's `shouldComponentUpdate` method:

https://github.com/marmelab/react-admin/blob/cd560cb9ba6f78efd098449e5e3febcf26418f25/packages/ra-input-rich-text/src/index.js#L61-L63

But when it executes, both values are the same, therefore component doesn't rerender:

Screenshot 2019-05-09 at 15 59 14

**Environment**

* React-admin version: 2.9.0
* ra-input-rich-text version: 2.8.6
* Last version that did not exhibit the issue (if applicable): N/A
* React version: N/A
* Browser: N/A
* Stack trace (in case of a JS error): N/A

### PR DESCRIPTION
**Pull request by cherniavskii**
As suggested in https://github.com/marmelab/react-admin/pull/3209#issuecomment-492273806, I've added tests.

Fixes #3206, #3207 and https://github.com/marmelab/react-admin/pull/3209#issuecomment-492246926

Closes #3209, #3221 and #3208 patches, since it contains them all.

### CODE REVIEW

File: cypress/integration/create.js [hunk 1]
```diff
@@ -212,4 +232,45 @@ describe('Create Page', () => {
...
+    it('should not show rich text input error message when field is untouched', () => {
+        cy.get('.ra-input-body .ra-rich-text-input > p').should('not.exist');
+    });
+
+    it('should show rich text input error message when form is submitted', () => {
+        CreatePage.submit();
+        cy.get('.ra-input-body .ra-rich-text-input > p').should('exist').contains('Required');
+    });
+
+    it.only('should not show rich text input error message when form is submitted and input is filled with text', () => {
```
Comment (djhi) [inline cypress/integration/create.js]: only spotted

Comment (cherniavskii) [reply cypress/integration/create.js]: Good catch!
BTW, is there better way to run single test in cypress?
Using `.only` has a drawback like this one

Comment (djhi) [reply cypress/integration/create.js]: None that I'm aware of. You miss jest too ? :wink:

Comment (cherniavskii) [reply cypress/integration/create.js]: Trying to use https://github.com/bahmutov/cypress-skip-and-only-ui , will submit a PR if it works as declared :)

**Comment by cherniavskii** [comment]
Let me know If there's anything blocking this PR from merging

File: cypress/integration/create.js [hunk 2]
```diff
@@ -212,4 +232,45 @@ describe('Create Page', () => {
             expect(el).to.have.value('The real Slim Shady!')
         );
     });
+
+    it('should not show rich text input error message when field is untouched', () => {
+        cy.get('.ra-input-body .ra-rich-text-input > p').should('not.exist');
```
Comment (djhi) [inline cypress/integration/create.js]: Looking for a `p` seems fragile. Would you mind adding a className (`error` ?) on the `FormHelperText` used to display validation errors?

Comment (cherniavskii) [reply cypress/integration/create.js]: Sure. done!

**Comment by cherniavskii** [comment]
@djhi can you rerun travis ci build? It's failed because of timeout error:

`CypressError: Timed out after waiting '60000ms' for your remote page to load.`

which seems to be caused by network issues. e2e tests are passing locally.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**